data-model/src/main/avro/Event.avdl

```diff
@@ -34,6 +34,8 @@ protocol EventProtocol {
     // Event may be associated with multiple entities
     array<string> entity_id_list = [];
 
+    int resource_index = -1;
```

Comment:
why is this index needed?

Reply:
This is the main point of the change. A span is associated with at most one resource - this index references that resource which is being held in the structured trace. Resources are the tracing equivalent to entities (which never should have been on the structured trace), but they're immutable.

Reply:
Resource will be reported alongside span message in the platform?

Reply:
Yes - see links to OLTP reporting format above (specifically, the ones in https://github.com/hypertrace/data-model/pull/16#discussion_r592805664 )